### PR TITLE
Add @types for ntlm-client

### PR DESCRIPTION
**Pull request by Borrajo**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/ntlm-client/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            SERVER: string;
+            DNS: string;
+            FQDN: string;
+            PARENT_DNS: string;
+        },
+        buffer: Buffer;
+    };
+}
+
+declare namespace ntlm {
```
Comment (jessetrinity) [inline types/ntlm-client/index.d.ts]: Does this need to be namespaced?

**Review by jessetrinity — CHANGES_REQUESTED** [review]

File: types/ntlm-client/index.d.ts (hunk 1 above)
Comment (jessetrinity) [reply types/ntlm-client/index.d.ts]: if you use 

```ts
import ntlm = require('ntlm-client');
```

in the test, you will see that you have to type
```ts
ntlm.ntlm.createType3Message
```
with the extra namespace layer.

**Comment by Borrajo** [comment]
Removed inneeded namespace.

File: types/ntlm-client/index.d.ts (hunk 1 above)
Comment (jessetrinity) [reply types/ntlm-client/index.d.ts]: Do you need the ntlm types as well? You can export those as interfaces:
https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html

File: types/ntlm-client/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// Type definitions for ntlm-client 0.1
+// Project: https://github.com/clncln1/node-ntlm-client
+// Definitions by: Manuel Borrajo <https://github.com/borrajo>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import "node";
```
Comment (sheetalkamat) [inline types/ntlm-client/index.d.ts]: This import is not needed.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/ntlm-client/index.d.ts (hunk 2 above)
Comment (Borrajo) [reply types/ntlm-client/index.d.ts]: if i remove the import, running the lint throws me this error:

`ERROR: 17:17  expect  TypeScript@next compile error: 
Cannot find name 'Buffer'. Do you need to install type definitions for node? Try npm i @types/node and then add node to the types field in your tsconfig.`

Then add 'node' in types in tsconfig 
`"types": [ "node" ]`
And when I run the lint again, error:
`Error: Expected compilerOptions["types"] === []`

So the only way I found to solve it was by `import "node"`. 
Can you think of any other way to solve this?

File: types/ntlm-client/index.d.ts (hunk 1 above)
Comment (Borrajo) [reply types/ntlm-client/index.d.ts]: That interface does not need to be exported, it is for internal use. The user has nothing to modify.

File: types/ntlm-client/index.d.ts (hunk 2 above)
Comment (sheetalkamat) [reply types/ntlm-client/index.d.ts]: If thats the case you need to change this to `/// <reference types="node"/>`

**Pull request merged**